tests/system_tests/wandb_backend_spy/spy.py

```diff
@@ -34,24 +32,61 @@ def freeze(self) -> Iterator[WandbBackendSnapshot]:
             finally:
                 snapshot._spy = None
 
-    def post_graphql(self, contents: bytes) -> fastapi.Response | None:
-        """Intercept a GraphQL request."""
-        return None
+    def post_graphql(
+        self,
+        request_raw: bytes,
+        response_raw: bytes,
+    ) -> None:
+        """Spy on a GraphQL request and response."""
+        request = json.loads(request_raw)
+
+        with self._lock:
+            query: str | None = request.get("query")
+            variables: dict[str, Any] | None = request.get("variables")
+            if query is None or variables is None:
+                return
+
+            print("GOT QUERY: ", query)
+            print("WITH VARIABLES: ", variables)
+
+            self._spy_run_config(query, variables)
+
+    def _spy_run_config(self, query: str, variables: dict[str, Any]) -> None:
+        """Detect changes to run config.
+
+        Requires self._lock.
+        """
+        # NOTE: This is an exact-string match to the query we send.
+        # It does not depend on the GraphQL schema.
+        if "mutation UpsertBucket" not in query:
+            return
+
+        if "config" not in variables:
+            return
+
+        run_id = variables["name"]
+        config = variables["config"]
+
+        print(f"UPDATED CONFIG {run_id}")
```

Comment:
hmm... probably want to remove the print?

Reply:
oh, oops, good catch